### DISCUSSION TITLE
Why did it take 180 milliseconds here？

### DISCUSSION THREAD

**Discussion by astrotzar** [milvus-io/milvus#18543] [Q&A and General discussion] [answered]
Why did it take 180 milliseconds here？Milvus-2.0.2 Single-node deployment, not SSD.
request:
 [timestamp=435072460913115138] [db=] [collection=alm_toc_goods_vector_data] [partitions="[toc_partition_4]"] [dsl=" shopId == 9991356 "] [len(PlaceholderGroup)=524] [OutputFields="[spuId,skuId,mchId,shopId,cat1,cat2,cat3]"] [search_params="[{\"key\":\"anns_field\",\"value\":\"vectorValue\"},{\"key\":\"topk\",\"value\":\"88\"},{\"key\":\"metric_type\",\"value\":\"L2\"},{\"key\":\"round_decimal\",\"value\":\"-1\"},{\"key\":\"params\",\"value\":\"{\\\"ef\\\":512}\"}]"] [travel_timestamp=0] [guarantee_timestamp=0]
![QA1](https://user-images.githubusercontent.com/3733700/183037151-2c290158-2441-4081-9eb9-61c486b55c08.jpg)

**Comment by yhmo** [comment, marked as answer]
In you search request, the parameter guarantee_timestamp is set to 0, so it is "Strong" consistency level search. Milvus will have 200ms latency for this kind of search to ensure data consistency.

For more information:
https://milvus.io/docs/v2.1.x/tune_consistency.md#Tune-consistency
https://github.com/milvus-io/milvus/blob/master/docs/developer_guides/how-guarantee-ts-works.md

You can try guarantee_timestamp=1, search request will be executed instantly.

**Reply by yhmo** [reply]
In Java SDK v2.0.2, use withGuaranteeTimestamp() to specify the guarantee_timestamp(set to 1 for instantly executing): 
```
SearchParam searchParam = SearchParam.newBuilder()
                .withCollectionName(COLLECTION_NAME)
                .withMetricType(MetricType.L2)
                .withOutFields(outFields)
                .withTopK(SEARCH_K)
                .withVectors(vectors)
                .withVectorFieldName(VECTOR_FIELD)
                .withExpr(expr)
                .withParams(SEARCH_PARAM)
                .withGuaranteeTimestamp(Constant.GUARANTEE_EVENTUALLY_TS)
                .build();

        R<SearchResults> response = milvusClient.search(searchParam);
```

**Reply by astrotzar** [reply]
Thank you very much!
